cli/command/stack/kubernetes/services.go

```diff
@@ -79,60 +78,47 @@ func getResourcesForServiceList(dockerCli *KubeCli, filters filters.Args, labelS
 	return replicas, daemons, services, nil
 }
 
-// RunServices is the kubernetes implementation of docker stack services
-func RunServices(dockerCli *KubeCli, opts options.Services) error {
+// GetServices is the kubernetes implementation of listing stack services
+func GetServices(dockerCli *KubeCli, opts options.Services) ([]swarm.Service, map[string]service.ListInfo, error) {
```

Comment:
I'm looking at the extra return here; the `service.ListInfo` is basically generated from information that's in `[]swarm.Service`, combined with information about the number of tasks running / desired. If I see it correctly, the `service.ListInfo` here is needed, because `swarm.Service` has no field for task status ("mode" is already part of the Service spec).

I _just_ (finally) merged https://github.com/moby/moby/pull/39231, which is a PR that adds exact those fields to the `Service` struct, which means we could just return `([]swarmService, error)`, and have the presentation code either convert it to a `ListInfo` or adjust that code to use `[]swarm.Service` directly.

Reply:
Let me prepare a vendor bump to bring in the changes from  https://github.com/moby/moby/pull/39231

Reply:
Sure, I can wait for the vendor bump, thanks!